modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuildController.java

```diff
@@ -141,12 +142,17 @@ private CompletableFuture<?> onIndexBuilding(StartBuildingIndexEventParameters p
 
             var startBuildIndexFutures = new ArrayList<CompletableFuture<?>>();
 
-            for (TablePartitionId primaryReplicaId : primaryReplicaIds) {
-                if (primaryReplicaId.tableId() == indexDescriptor.tableId()) {
-                    CompletableFuture<?> startBuildIndexFuture = getMvTableStorageFuture(parameters.causalityToken(), primaryReplicaId)
-                            .thenCompose(mvTableStorage -> awaitPrimaryReplica(primaryReplicaId, clockService.now())
+            for (ZonePartitionId zonePartitionId : primaryReplicaIds) {
+
+                int tableId = zonePartitionId.tableId();
+
+                TablePartitionId tablePartId = new TablePartitionId(tableId, zonePartitionId.partitionId());
+
+                if (tableId == indexDescriptor.tableId()) {
+                    CompletableFuture<?> startBuildIndexFuture = getMvTableStorageFuture(parameters.causalityToken(), tablePartId)
+                            .thenCompose(mvTableStorage -> awaitPrimaryReplica(zonePartitionId, clockService.now())
```

Comment:
because it is `IndexBuildController#awaitPrimaryReplica`, not `PlacementDriver#awaitPrimaryReplicaForTable`. Or maybe I didn't understand your question and you asked for renaming `IndexBuildController#awaitPrimaryReplica`?

Reply:
No, I've just missed the fact that it's `IndexBuildController#awaitPrimaryReplica`.